airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry_entry.py

```diff
@@ -213,6 +212,7 @@ def persist_registry_entry_to_json(
     registry_entry_write_path = get_registry_entry_write_path(metadata_entry, registry_name)
     registry_entry_json = registry_entry.json(exclude_none=True)
     file_handle = registry_directory_manager.write_data(registry_entry_json.encode("utf-8"), ext="json", key=registry_entry_write_path)
+    HACKS.write_registry_to_overrode_file_paths(registry_entry, registry_name, metadata_entry, registry_directory_manager)
```

Comment:
Q: This means when publishing a pre-release, will we need to publish a pre-release for both the regular _and_ strict-encrypt variants?

I ask because I _think_ this reads as "when persisting a registry entry, also persist it in this other location", but in the case of pre-releases for only the strict_encrypt no registry entry is generated in the first place

Reply:
@pedroslopez nope!

This just means that if we find a registry entry file (oss.json, cloud.json) where the dockerRepository has been overriden then we make sure to also write it to
`metadata-bucket/metadata/{overridden-dockerrepo}/{version}/cloud.json`

Reply:
But that does mean we are in a weird spot for when to and when not to publish the strict encrypt variant

Reply:
I dont think we have a good short term way around this.

But at the same time I think its good that the organization feel a bit of pain around these strict encrypt variants

Reply:
@bnchrch yes, but in the case of pre-releases no registry entry file is in place for the non-strict-encrypt variant because we don't tend to publish pre-releases for both (at least not historically).

For example, see source-postgres-strict-encryppt:
<img width="316" alt="image" src="https://github.com/airbytehq/airbyte/assets/4368928/2c40c550-137b-4935-b79c-0d33ce9bf330">

and source-postgres

<img width="303" alt="image" src="https://github.com/airbytehq/airbyte/assets/4368928/06c83089-c872-49ba-ba2c-80134eb74c19">


The pre-releases that are in there are different because people just usually dont publish both

Reply:
With this change, I think the process now would be:

1. publish pre-relese for strict-encrypt
2. publish pre-release for regular variant so the registry entry is made for it and copied over to strict-encrypt

Does that make sense?: Am I missing something?

Reply:
@pedroslopez Makes sense! Looks like I published my edit at the same time you published your explanation 😅 

https://github.com/airbytehq/airbyte/pull/28891#discussion_r1280031645

Reply:
We just have a failing test. Going to fix that up then merge